src/redux/actions/experiments/createExperiment.js

```diff
@@ -1,47 +1,99 @@
 /* eslint-disable no-param-reassign */
+import fetchAPI from 'utils/fetchAPI';
 import moment from 'moment';
 import hash from 'object-hash';
 
-import saveExperiment from './saveExperiment';
+import { api } from 'utils/constants';
+
 import {
   EXPERIMENTS_CREATED,
   EXPERIMENTS_ERROR,
-} from '../../actionTypes/experiments';
-import { experimentTemplate } from '../../reducers/experiments/initialState';
-import endUserMessages from '../../../utils/endUserMessages';
-import pushNotificationMessage from '../../../utils/pushNotificationMessage';
+  EXPERIMENTS_SAVING,
+  EXPERIMENTS_SAVED,
+} from 'redux/actionTypes/experiments';
+import { experimentTemplate } from 'redux/reducers/experiments/initialState';
+
+import endUserMessages from 'utils/endUserMessages';
+import pushNotificationMessage from 'utils/pushNotificationMessage';
+import { isServerError, throwIfRequestFailed } from 'utils/fetchErrors';
+import experimentConvertedToApiModel from 'utils/convertExperimentToApiModel';
 
 const createExperiment = (
   projectUuid, newExperimentName,
-) => async (dispatch) => {
+) => async (dispatch, getState) => {
   const createdDate = moment().toISOString();
 
+  const experimentId = hash.MD5(createdDate);
+
   const newExperiment = {
     ...experimentTemplate,
-    id: hash.MD5(createdDate),
+    id: experimentId,
     name: newExperimentName,
     projectUuid,
     createdDate,
   };
 
+  let url;
+  let experimentToSend;
+
   try {
-    dispatch(saveExperiment(newExperiment.id, newExperiment, false));
+    if (api.CURRENT_VERSION === api.possibleVersions.V1) {
+      experimentToSend = newExperiment || getState().experiments[experimentId];
+      experimentToSend = experimentConvertedToApiModel(experimentToSend);
+
+      url = `/v1/experiments/${experimentId}`;
+    } else if (api.CURRENT_VERSION === api.possibleVersions.V2) {
+      const { id, name, description } = newExperiment || getState().experiments[experimentId];
+      experimentToSend = { id, name, description };
+
+      url = `/v2/experiments/${experimentId}`;
+    }
+
+    dispatch({
+      type: EXPERIMENTS_SAVING,
+    });
+
+    const response = await fetchAPI(
+      url,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(experimentToSend),
+      },
+    );
+
+    const json = await response.json();
+    throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
 
     dispatch({
       type: EXPERIMENTS_CREATED,
       payload: {
         experiment: newExperiment,
       },
     });
-  } catch (e) {
-    pushNotificationMessage('error', endUserMessages.ERROR_SAVING);
 
+    dispatch({
+      type: EXPERIMENTS_SAVED,
+    });
```

Comment:
why do we need to dispatch these two redux actions in a row? Is there any situation when we need to dispatch only one of them, but not the other? If there is no such situation, I suggest that we combine them into 1 action.